spec/services/replication_transfer_updater_spec.rb

```diff
@@ -0,0 +1,266 @@
+        it_behaves_like "store_requested_update"
+        it "does not request preservation" do
+          bag_man_request = double(:bag_man_request)
+          allow(record).to receive(:bag_man_request).and_return(bag_man_request)
+          expect(bag_man_request).to_not receive(:okay_to_preserve!)
+          update.update
+        end
+      end
+      context "we are replicating" do
+        let(:record) { Fabricate(:replication_transfer, store_requested: false, to_node: Fabricate(:local_node)) }
+        it_behaves_like "store_requested_update"
+        it "requests preservation" do
+          bag_man_request = double(:bag_man_request)
+          allow(record).to receive(:bag_man_request).and_return(bag_man_request)
+          expect(bag_man_request).to receive(:okay_to_preserve!)
+          update.update
+        end
+      end
+    end
+  end
+
+
+  describe ReplicationTransferUpdater::StoredUpdate do
+    let(:update) { described_class.new(record, updated) }
+    let(:changes) { {stored: true} }
+    describe "::matching_update?" do
+      let(:record) { Fabricate(:replication_transfer, stored: false) }
+      it "matches old.stored == false, new.stored == true" do
+        expect(described_class.matching_update?(record, updated)).to be true
+      end
+    end
+    describe "#update" do
+      context "we did not request this transfer" do
+        let(:record) { Fabricate(:replication_transfer, stored: false) }
+        it "updates the record" do
+          update.update
+          expect(record.stored).to be true
+        end
+        it "returns true on success" do
+          allow(record).to receive(:update).and_return true
+          expect(update.update).to be true
+        end
+        it "returns false on failure" do
+          allow(record).to receive(:update).and_return false
+          expect(update.update).to be false
+        end
+        it "does not cancel the record" do
+          update.update
+          expect(record.cancelled).to be false
+        end
+        it "does not modify the bag's replicating nodes" do
+          update.update
+          expect(record.bag.replicating_nodes).to_not include(record.to_node)
+        end
+      end
+      context "we requested this transfer" do
+        let(:record) { Fabricate(:replication_transfer, stored: false, from_node: Fabricate(:local_node)) }
+        it "updates the record" do
+          update.update
+          expect(record.stored).to be true
+        end
+        it "returns true on success" do
+          allow(record).to receive(:update).and_return true
+          expect(update.update).to be true
+        end
+        it "returns false on failure" do
+          allow(record).to receive(:update).and_return false
+          expect(update.update).to be false
+        end
+        it "does not cancel the record" do
+          update.update
+          expect(record.cancelled).to be false
+        end
```

Comment:
Pull out lines 245-253 into a global shared context so they can be replaced here and elsewhere with a reference to that shared context.